packages/devtools_app/test/vm_developer/object_inspector/object_viewport_test.dart

```diff
@@ -0,0 +1,276 @@
+    final fakeScriptRef = ScriptRef(uri: 'foo.dart', id: '1234');
+    final testScriptObject =
+        TestScriptObject(ref: fakeScriptRef, testScript: fakeScript);
+    testObjectInspectorViewController.fakeObjectHistory
+        .setCurrentObject(testScriptObject);
+    await tester.pumpWidget(
+      wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+    );
+    expect(viewportTitle(testScriptObject), 'Script @ foo.dart');
+    expect(find.text('Script @ foo.dart'), findsOneWidget);
+    expect(find.bySubtype<VMInfoCard>(), findsOneWidget);
+  });
+
+  testWidgets('test for Field Object', (WidgetTester tester) async {
+    final testFieldObject =
+        TestFieldObject(ref: fakeFieldRef, testField: fakeField);
+    testObjectInspectorViewController.fakeObjectHistory
+        .setCurrentObject(testFieldObject);
+    await tester.pumpWidget(
+      wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+    );
+    expect(viewportTitle(testFieldObject), 'Field FooField');
+    expect(find.text('Field FooField'), findsOneWidget);
+    expect(find.bySubtype<VMInfoCard>(), findsOneWidget);
+  });
+
+  testWidgets('test for Library Object', (WidgetTester tester) async {
+    final testLibraryObject =
+        TestLibraryObject(ref: fakeLibRef, testLibrary: fakeLib);
+    testObjectInspectorViewController.fakeObjectHistory
+        .setCurrentObject(testLibraryObject);
+    await tester.pumpWidget(
+      wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+    );
+    expect(viewportTitle(testLibraryObject), 'Library FooLib');
+    expect(find.text('Library FooLib'), findsOneWidget);
+    expect(find.bySubtype<VMInfoCard>(), findsOneWidget);
+  });
+
+  testWidgets('test for Instance Object', (WidgetTester tester) async {
+    final testInstanceObject =
+        TestInstanceObject(ref: fakeInstanceRef, testInstance: fakeInstance);
+    testObjectInspectorViewController.fakeObjectHistory
+        .setCurrentObject(testInstanceObject);
+    await tester.pumpWidget(
+      wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+    );
+    expect(viewportTitle(testInstanceObject), 'Instance FooInstance');
+    expect(find.text('Instance FooInstance'), findsOneWidget);
+    expect(find.bySubtype<VMInfoCard>(), findsOneWidget);
+  });
+
+  testWidgets('test for Func Object', (WidgetTester tester) async {
+    final testFuncObject =
+        TestFuncObject(ref: fakeFunctionRef, testFunc: fakeFunction);
+    testObjectInspectorViewController.fakeObjectHistory
+        .setCurrentObject(testFuncObject);
+    await tester.pumpWidget(
+      wrap(ObjectViewport(controller: testObjectInspectorViewController)),
+    );
+    expect(viewportTitle(testFuncObject), 'Function FooFunction');
+    expect(find.text('Function FooFunction'), findsOneWidget);
+    expect(find.bySubtype<VMInfoCard>(), findsOneWidget);
+  });
+
+  group('ObjectHistory', () {
```

Comment:
the only tests you need in this group are the ones that test the method you added to ObjectHistory (`pushEntry`). Some of these tests are duplicated and already tested in history_manager_test.dart, so no need to copy them here.

Reply:
Done.
I did notice when testing pop, that it gives an error when popping an item when it is the only item in the stack. Because current does not go back to null, but instead tries to access element at index -1.

Reply:
In that case, we should add the pop test to history_manager_test.dart and fix that case so that we correctly set to null👍

Reply:
Should that I fix that in a separate PR or on this one?

Reply:
Whichever you prefer. Normally I'd say separate PR, but I don't remind reviewing here to speed it along and avoid merge conflicts for you.